src/main/kotlin/no/nav/klage/oppgave/service/VedtakService.kt

```diff
@@ -158,22 +148,86 @@ class VedtakService(
         }
     }
 
-    fun dispatchVedtakToKafka(klagebehandlingId: UUID, vedtakId: UUID) {
-        val klage = klagebehandlingRepository.findById(klagebehandlingId).orElseThrow()
-        val vedtak = klage.vedtak.find { it.id == vedtakId }
-        require(vedtak != null) { "Fant ikke vedtak på klage" }
+    fun finalizeVedtak(
+        klagebehandlingId: UUID,
+        vedtakId: UUID,
+        input: VedtakFullfoerInput,
+        innloggetIdent: String
+    ) {
+        val klage = klagebehandlingService.getKlagebehandlingForUpdate(
+            klagebehandlingId,
+            input.klagebehandlingVersjon
+        )
+        val vedtak = getVedtakFromKlagebehandling(klage, vedtakId)
+        if (vedtak.finalized != null) throw VedtakFinalizedException("Vedtak med id $vedtakId er allerede ferdigstilt")
+        if (vedtak.journalpostId == null) throw JournalpostNotFoundException("Vedtak med id $vedtakId er ikke journalført")
```

Comment:
Her gjøres tilsynelatende det samme på to ulike måter. To if-sjekker, og så etterpå en "require" ? Kunne det vært gjort på samme måte alle tre tingene?